crowdsec-docs/unversioned/bouncers/haproxy_spoa.mdx

```diff
@@ -0,0 +1,400 @@
+---
+id: haproxy_spoa
+title: HAProxy SPOA
+sidebar_position: 10
+---
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+import useBaseUrl from '@docusaurus/useBaseUrl';
+import RemediationSupportBadges from '@site/src/components/RemediationSupportBadges.tsx';
+
+<p align="center">
+<img src={useBaseUrl('/img/crowdsec_haproxy.svg')} alt="CrowdSec" title="CrowdSec" width="400" height="300" />
+</p>
+<p align="center">
+<img src="https://img.shields.io/badge/build-pass-green" />
+<img src="https://img.shields.io/badge/tests-pass-green" />
+</p>
+<p align="center">
+&#x1F4DA; <a href="#installation/">Documentation</a>
+&#x1F4A0; <a href="https://hub.crowdsec.net">Hub</a>
+&#128172; <a href="https://discourse.crowdsec.net">Discourse </a>
+</p>
+
+<RemediationSupportBadges
+  Appsec={false}
+  Metrics
+  Prometheus
+/>
+
+
+# A Remediation Component for haproxy.
+
+
+## What it does ?
+
+The `cs-haproxy-spoa-bouncer` allows CrowdSec to enforce blocking, CAPTCHA, or
+allow actions directly within HAProxy using the [SPOE
+protocol](https://www.haproxy.com/blog/extending-haproxy-with-the-stream-processing-offload-engine).
+
+This remediation component is meant to obsolete the old lua-based haproxy bouncer.
+
+It supports IP-based decisions, CAPTCHA challenges, GeoIP-based headers, and
+integrates cleanly with CrowdSec’s LAPI using the stream bouncer protocol.
+
+Supported features:
+
+ - Stream mode (pull the local API for new/old decisions every X seconds)
+ - Ban remediation (can ban an IP address by redirecting or returning a custom HTML page)
+ - Captcha remediation (can return a captcha)
+ - Works with IPv4/IPv6
+ - Support IP ranges (can apply a remediation on an IP range)
+ - We are working on supporting AppSec
+
+
+## Installation
```

Comment:
I think somewhere in the install we should point out "how to check that it's correctly running" either by viewing the metrics or so.